packages/transport-webrtc/src/private-to-public/utils/certificate-store.ts

```diff
@@ -0,0 +1,151 @@
+
+export interface CertificateStoreOptions {
+  validityDays?: number
+  minRemainingValidityDays?: number
+}
+
+/**
+ * Check if a stored certificate exists and is valid
+ */
+export async function getStoredCertificate (keychain: Keychain, 
+                                           options: CertificateStoreOptions = {}, 
+                                           log?: Logger): Promise<TransportCertificate | null> {
+  const minRemainingDays = options.minRemainingValidityDays ?? DEFAULT_MIN_REMAINING_VALIDITY_DAYS
+  
+  try {
+    // Try to find an existing certificate
+    const keyName = CERTIFICATE_KEY_PREFIX
+    
+    // Export the certificate key material
+    const privateKey = await keychain.exportKey(keyName)
+    
+    // Parse the stored certificate format to extract metadata
+    const parsedCert = await importPrivateKey(privateKey)
+    
+    if (parsedCert == null) {
+      log?.trace('stored certificate could not be parsed')
+      return null
+    }
+    
+    // Check if the certificate is still valid with sufficient remaining time
+    const now = new Date()
+    const minValidUntil = new Date(now)
+    minValidUntil.setDate(minValidUntil.getDate() + minRemainingDays)
+    
+    if (parsedCert.expiryDate > minValidUntil) {
+      log?.trace(`found valid certificate (expires ${parsedCert.expiryDate.toISOString()})`)
+      return parsedCert.certificate
+    }
+    
+    log?.trace(`certificate expires too soon (${parsedCert.expiryDate.toISOString()})`)
+    return null
+  } catch (err) {
+    // If certificate doesn't exist or has an issue, return null
+    log?.trace('failed to retrieve stored certificate', err)
+    return null
+  }
+}
+
+/**
+ * Store a certificate in the keychain
+ */
+export async function storeCertificate (keychain: Keychain, certificate: TransportCertificate, log?: Logger): Promise<void> {
+  try {
+    // Create a suitable key for the keychain from our certificate
+    const certKey = await createPrivateKeyFromCertificate(certificate)
+    
+    try {
+      // Try to remove any existing certificate before storing the new one
+      await keychain.removeKey(CERTIFICATE_KEY_PREFIX)
+    } catch (err) {
+      // Ignore errors if the key doesn't exist
+    }
+    
+    // Store the new certificate
+    await keychain.importKey(CERTIFICATE_KEY_PREFIX, certKey)
+    log?.trace('successfully stored certificate in keychain')
+  } catch (err) {
+    log?.error('failed to store certificate in keychain', err)
+    throw err
+  }
+}
+
+/**
+ * Generate a new transport certificate and optionally store it in the keychain
+ */
+export async function generateAndStoreCertificate (
+  keychain?: Keychain,
+  options: CertificateStoreOptions = {},
+  log?: Logger
+): Promise<TransportCertificate> {
+  const validityDays = options.validityDays ?? DEFAULT_CERTIFICATE_VALIDITY_DAYS
+  
+  log?.trace('generating new ECDSA P-256 key pair')
+  const keyPair = await crypto.subtle.generateKey({
```

Comment:
You can now import `generateKeyPair` from `@libp2p/crypto/keys` and do `await generateKeyPair('ECDSA')`.

Then the keypair can be stored in the keychain and you can use `privateKeyToCryptoKeyPair` (also imported from `@libp2p/crypto/keys`) to get a WebCrypto-friendly keypair to generate the cert.